Refactor

mirovix opened the pull request:
1. Unit Test completed
2. Integration Test completed
3. Refactoring

mirovix committed:
feat: unit test creation for the filter

mirovix committed:
chore: chaning layout

ltonin commented on CMakeLists.txt:
```diff
@@ -28,14 +29,18 @@ catkin_package(
 include_directories(${catkin_INCLUDE_DIRS} 
                     ${EIGEN3_INCLUDE_DIRS}
-					include)
+					include/${PROJECT_NAME}/)
```
include/${PROJECT_NAME}/ cannot find the proper include directory.

The correct way is:
include_directories(${catkin_INCLUDE_DIRS}  ${EIGEN3_INCLUDE_DIRS}
					include)

mirovix committed:
fix: changing directories

The correct way is:
include_directories(${catkin_INCLUDE_DIRS} ${EIGEN3_INCLUDE_DIRS}
include)

mirovix commented:
Fixed directories to include
